cli/js/files.ts

```diff
@@ -231,5 +281,82 @@ export type OpenMode =
  * @internal
  */
 export function create(filename: string): Promise<File> {
-  return open(filename, "w+");
+  return open(filename, {
+    read: true,
+    create: true,
+    write: true,
+    truncate: true
+  });
+}
+
+/** Check if OpenMode is set to something that is valid.
+ *  @returns Tuple representing if openMode is valid and error message if it's not
+ *  @internal
+ */
+function checkOpenMode(mode: OpenMode): [boolean, string] {
+  const allOptionsAreFalse =
+    Object.values(mode).filter(val => val == true).length === 0;
+  const truncateOptionWithoutWriteAccess = mode.truncate && !mode.write;
+  const createOrCreateNewWithoutWriteOrAppend =
+    (mode.create || mode.createNew) && !(mode.write || mode.append);
+  if (allOptionsAreFalse)
+    return [false, "OpenMode require at least one option to be true"];
+  if (truncateOptionWithoutWriteAccess)
+    return [false, "Truncate option require write accesss"];
+  if (createOrCreateNewWithoutWriteOrAppend)
+    return [false, "Create or create new requires write or append access"];
+  return [true, ""];
+}
+
+/** Converts OpenModeLegacy to Legacy
+ *  @internal
+ */
+function convertOpenModeLegacy(mode: OpenModeLegacy): OpenMode {
+  const r = {
+    read: true
+  };
+  const rplus = {
+    read: true,
+    write: true
+  };
+  const w = {
+    write: true,
+    truncate: true,
+    create: true
+  };
+  const wplus = {
+    write: true,
+    truncate: true,
+    create: true,
+    read: true
+  };
+  const a = {
+    append: true,
+    create: true
+  };
+  const aplus = {
+    append: true,
+    create: true,
+    read: true
+  };
+  const x = {
+    write: true,
+    createNew: true
+  };
+  const xplus = {
+    write: true,
+    createNew: true,
+    read: true
+  };
+
+  const map = {};
+  map["r"] = r;
+  map["r+"] = rplus;
+  map["w"] = w;
+  map["w+"] = wplus;
+  map["a"] = a;
+  map["a+"] = aplus;
+  map["x"] = x;
+  map["x+"] = xplus;
```

Comment:
I think this can all be shortened up and made more readable by not declaring variables, but rather

```
const legacyMap = {
  "r": { read: true},
  "w": { write: true},
  // ...
};

function convertOpenModeLegacy(mode: OpenModeLegacy): OpenMode {
  return legacyMap[mode];
}

```

Reply:
yes, that's better!

however, as suggested I decided to remove legacy open mode completely as we aren't even in 1.0

Reply:
I just read your comment @ry above.

So - keep legacy or not?